stores/yaml/store.go

```diff
@@ -118,45 +122,89 @@ func (store *Store) LoadEncryptedFile(in []byte) (sops.Tree, error) {
 		return sops.Tree{}, err
 	}
 	// After that, we load the whole file into a map.
-	var data yaml.MapSlice
-	if err := (yaml.CommentUnmarshaler{}).Unmarshal(in, &data); err != nil {
-		return sops.Tree{}, fmt.Errorf("Error unmarshaling input YAML: %s", err)
-	}
 	// Discard metadata, as we already loaded it.
-	for i, item := range data {
+	for i, item := range data[0] {
 		if item.Key == "sops" {
-			data = append(data[:i], data[i+1:]...)
+			data[0] = append(data[0][:i], data[0][i+1:]...)
 		}
 	}
+
+	var branches sops.TreeBranches
+	for _, doc := range data {
+		for i, item := range doc {
+			if item.Key == "sops" { // Erase
+				doc = append(doc[:i], doc[i+1:]...)
+			}
+		}
+		branches = append(branches, store.mapSliceToTreeBranch(doc))
+	}
 	return sops.Tree{
-		Branch:   store.mapSliceToTreeBranch(data),
+		Branches: branches,
 		Metadata: metadata,
 	}, nil
 }
 
-func (store *Store) LoadPlainFile(in []byte) (sops.TreeBranch, error) {
-	var data yaml.MapSlice
-	if err := (yaml.CommentUnmarshaler{}).Unmarshal(in, &data); err != nil {
+func (store *Store) LoadPlainFile(in []byte) (sops.TreeBranches, error) {
+	var data []yaml.MapSlice
+	if err := (yaml.CommentUnmarshaler{}).UnmarshalDocuments(in, &data); err != nil {
 		return nil, fmt.Errorf("Error unmarshaling input YAML: %s", err)
 	}
-	return store.mapSliceToTreeBranch(data), nil
+
+	var branches sops.TreeBranches
+	for _, doc := range data {
+		branches = append(branches, store.mapSliceToTreeBranch(doc))
+	}
+	return branches, nil
 }
 
 func (store *Store) EmitEncryptedFile(in sops.Tree) ([]byte, error) {
-	yamlMap := store.treeBranchToYamlMap(in.Branch)
-	yamlMap = append(yamlMap, yaml.MapItem{Key: "sops", Value: stores.MetadataFromInternal(in.Metadata)})
-	out, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
-	if err != nil {
-		return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+	out := []byte{}
+	var err error
+	if len(in.Branches) > 1 {
+		for i, branch := range in.Branches {
+			if i > 0 {
+				out = append(out, "---\n"...)
+			}
+			yamlMap := store.treeBranchToYamlMap(branch)
+			yamlMap = append(yamlMap, yaml.MapItem{Key: "sops", Value: stores.MetadataFromInternal(in.Metadata)})
+			tout, err := (&yaml.YAMLMarshaler{Indent: 4}).Marshal(yamlMap)
+			if err != nil {
+				return nil, fmt.Errorf("Error marshaling to yaml: %s", err)
+			}
+			out = append(out, tout...)
+		}
+	} else {
```

Comment:
Why is this `if ... else` needed? The loop you wrote would just emit the first branch and not add the  `---\n`, right? I don't think we need to special-case the 'single-branch' scenario with how you wrote the code.

Reply:
ah looks like I just missed this in the refactor. thanks for the catch, I'll fix.